actors/builtin/miner/miner_test.go

```diff
@@ -948,6 +948,127 @@ func TestCommitments(t *testing.T) {
+			dlInfo.NextNotElapsed().Last(): {pIdx},
+			quantizedExpiration:            {pIdx},
+		}, dQueue)
+
+		pQueue := actor.collectPartitionExpirations(rt, partition)
+		assertBitfieldEquals(t, pQueue[dlInfo.NextNotElapsed().Last()].OnTimeSectors, uint64(oldSector.SectorNumber))
+		assertBitfieldEquals(t, pQueue[quantizedExpiration].OnTimeSectors, uint64(newSector.SectorNumber))
+
+		// now declare old sector faulty
+		actor.declareFaults(rt, oldSector)
+
+		deadline, _ = actor.getDeadlineAndPartition(rt, dlIdx, partIdx)
+		dQueue = actor.collectDeadlineExpirations(rt, deadline)
+		dlInfo = miner.NewDeadlineInfo(st.ProvingPeriodStart, dlIdx, rt.Epoch())
+		expectedReplacedExpiration := dlInfo.QuantSpec().QuantizeUp(rt.Epoch() + miner.FaultMaxAge)
+		quantizedExpiration = dlInfo.QuantSpec().QuantizeUp(oldSector.Expiration)
+
+		// deadling marks expirations for partition at expiration epoch
+		assert.Equal(t, map[abi.ChainEpoch][]uint64{
+			dlInfo.NextNotElapsed().Last(): {pIdx},
+			expectedReplacedExpiration:     {pIdx},
+			quantizedExpiration:            {pIdx},
+		}, dQueue)
+
+		// but partitions expiration set at that epoch is empty
+		_, partition = actor.getDeadlineAndPartition(rt, dlInfo.Index, pIdx)
+		queue, err := miner.LoadExpirationQueue(rt.AdtStore(), partition.ExpirationsEpochs, dlInfo.QuantSpec())
+		require.NoError(t, err)
+		var es miner.ExpirationSet
+		expirationSetNotEmpty, err := queue.Get(uint64(expectedReplacedExpiration), &es)
+		require.NoError(t, err)
+		assert.False(t, expirationSetNotEmpty)
+
+		// advance to sector proving deadline
+		dlInfo = actor.deadline(rt)
+		for dlIdx != dlInfo.Index {
+			advanceDeadline(rt, actor, &cronConfig{})
+			dlInfo = actor.deadline(rt)
+		}
+
+		// submit post for new sector
+		rt.SetEpoch(dlInfo.Last())
+		partitions := []miner.PoStPartition{
+			{Index: pIdx, Skipped: bitfield.New()},
+		}
+		// oldSector is faulty, so expect newSector twice in validation; once for its proof and once to replace oldSector
+		// power is added for new sector and no penalties are paid yet
+		newPower := miner.QAPowerForSector(actor.sectorSize, newSector)
+		actor.submitWindowPoSt(rt, dlInfo, partitions, []*miner.SectorOnChainInfo{newSector, newSector}, &poStConfig{
+			expectedPowerDelta: miner.NewPowerPair(big.NewInt(int64(actor.sectorSize)), newPower),
+			expectedPenalty:    big.Zero(),
+		})
+
+		// At proving period cron expect to pay declared fee for new sector
+		// and to have its pledge requirement deducted indicating it has expired.
+		//oldPower := miner.QAPowerForSector(actor.sectorSize, oldSector)
+		//expectedPowerDelta := miner.NewPowerPair(big.NewInt(int64(actor.sectorSize)), oldPower).Neg()
+		//expectedFee := miner.PledgePenaltyForDeclaredFault(actor.epochRewardSmooth, actor.epochQAPowerSmooth, oldPower)
+		//actor.applyRewards(rt, expectedFee, big.Zero())
+		actor.onDeadlineCron(rt, &cronConfig{
+			//detectedFaultsPenalty:     expectedFee,
```

Comment:
I think the fix is to revert my ill-advised comment [here](https://github.com/filecoin-project/specs-actors/pull/1009#discussion_r475382622).